source/MaterialXRender/ImageHandler.h

```diff
@@ -258,6 +258,13 @@ class ImageHandler
         return _searchPath;
     }
 
+    /// Returns the bound texture location for a given resource
+    virtual int getBoundTextureLocation(unsigned int)
+    {
+        return 0;
```

Comment:
You have inconsistent logic. I think this should return -1 since you check for < 0 below for validity.